src/ert/analysis/_es_update.py

```diff
@@ -131,6 +132,28 @@ def _create_temporary_parameter_storage(
     return temporary_storage
 
 
+def _correlated_parameter_response_pairs(
+    A: npt.NDArray[np.float_], Y: npt.NDArray[np.float_], correlation_threshold: float
+) -> npt.NDArray[np.int_]:
+    N = A.shape[1]
+    Y_prime = Y - Y.mean(axis=1, keepdims=True)
+    C_YY = Y_prime @ Y_prime.T / (N - 1)
+    Sigma_Y = np.diag(np.sqrt(np.diag(C_YY)))
+
+    A_prime = A - A.mean(axis=1, keepdims=True)
+    C_AA = A_prime @ A_prime.T / (N - 1)
```

Comment:
Are you computing the full error covariance matrix C_AA of A here? This one can become huge, and you only need the diagonal which you can compute with N inner products of each of the N columns of A with themself.

Reply:
I might be misunderstanding something but `A_prime` is here a `1 by N` vector because we are looping over all parameters so `C_AA` is not large.
Were you thinking of something else perhaps?

Reply:
Sorry, then I misunderstood (or didn't look carefully enough).